components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/service/APIKeyMgtSubscriberService.java

```diff
@@ -208,12 +209,18 @@ public OAuthApplicationInfo createOAuthApplicationByApplicationInfo(OAuthApplica
 
             oAuthConsumerAppDTO.setOAuthVersion(OAuthConstants.OAuthVersions.VERSION_2);
             log.debug("Creating OAuth App " + applicationName);
-            oAuthAdminService.registerOAuthApplicationData(oAuthConsumerAppDTO);
+            OAuthConsumerAppDTO createdApp;
+            boolean isHashDisabled = OAuth2Util.isHashDisabled();
+            if (isHashDisabled) {
```

Comment:
why it's required to go through via two approaches?

Reply:
Since we are hashing the client secret value, we can't get the original value of the client secret (Not hashed one) in order to show it in the UI.

Here, after registering the new OAuth app [1] & [2] the hashed client secret value will be saved in the database. In order to show the Oauth application details(Client secret) in the UI, we are getting those from the database [4]. So, In the UI we will get the hashed value [5]. For this, we need to keep the original value of the client secret. Keeping this value in the Service context/ some where else is not a good approach. So, I have introduced to a new method that will return OAuthApp details after registering the new application. So that we can get the original value of the client secret value at [1].


[1] https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/v5.3.4/components/org.wso2.carbon.identity.oauth.ui/src/main/resources/web/oauth/add-finish-ajaxprocessor.jsp#L104
[2] https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/v5.3.4/components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminService.java#L207

[3] https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/v5.3.4/components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminService.java#L278

[4] https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/v5.3.4/components/org.wso2.carbon.identity.oauth.ui/src/main/resources/web/oauth/add-finish-ajaxprocessor.jsp#L106

[5] https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/v5.3.4/components/org.wso2.carbon.identity.oauth.ui/src/main/resources/web/oauth/add-finish-ajaxprocessor.jsp#L125